src/pages/debrief/cat-b/debrief.cat-b.selector.ts

```diff
@@ -0,0 +1,214 @@
+import { forOwn, transform, endsWith } from 'lodash';
+import { CatBUniqueTypes } from '@dvsa/mes-test-schema/categories/B';
+import {
+  manoeuvreTypeLabels,
+  manoeuvreCompetencyLabels,
+} from '../../../shared/constants/competencies/catb-manoeuvres';
+import { ManoeuvreTypes } from '../../../modules/tests/test-data/test-data.constants';
+import { CompetencyOutcome } from '../../../shared/models/competency-outcome';
+import {
+  MultiFaultAssignableCompetency,
+  CommentedCompetency,
+  CommentSource,
+} from '../../../shared/models/fault-marking.model';
+import { CompetencyDisplayName } from '../../../shared/models/competency-display-name';
+import { getSeriousOrDangerousFaults } from '../debrief.selector';
+// TODO - A lot of these are used in multiple places (Debrief, View Test Result, Office),
+// should be refactored into a more common area.
+export const getManoeuvreFaults = (
+  manoeuvres: CatBUniqueTypes.Manoeuvres,
+  faultType: CompetencyOutcome,
+): (CommentedCompetency & MultiFaultAssignableCompetency)[] => {
+  const faultsEncountered: (CommentedCompetency & MultiFaultAssignableCompetency)[] = [];
+  forOwn(manoeuvres, (manoeuvre, type: ManoeuvreTypes) => {
+    let competencyComment = '';
+    const faults = !manoeuvre.selected ? [] : transform(manoeuvre, (result, value, key: string) => {
+      if (endsWith(key, 'Fault') && value === faultType) {
+        if (key === 'controlFault') {
+          competencyComment = manoeuvre.controlFaultComments || '';
+        } else {
+          competencyComment = manoeuvre.observationFaultComments || '';
+        }
+        const manoeuvreFaultSummary: CommentedCompetency & MultiFaultAssignableCompetency = {
+          comment: competencyComment || '',
+          competencyIdentifier: `${type}${manoeuvreCompetencyLabels[key]}` ,
+          competencyDisplayName: [manoeuvreTypeLabels[type], manoeuvreCompetencyLabels[key]].join(' - '),
+          source: `${CommentSource.MANOEUVRES}-${type}-${manoeuvreCompetencyLabels[key]}`,
+          faultCount: 1,
+        };
+        result.push(manoeuvreFaultSummary);
+      }
+    }, []);
+    faultsEncountered.push(...faults);
+  });
+  return faultsEncountered;
+};
+
+export const getManoeuvreFaultsCount = (
```

Comment:
Maybe some duplication here with the `sumManoeuvreFaults` method from the `FaultCountProvider`

Reply:
This PR is about taking what is already there and making it CatBE aware.  Currently the FaultCount provider doesn't work properly for Cat BE so will refactor in another PR.